support/android/apk/app/src/main/java/com/mozilla/servo/MainActivity.java

```diff
@@ -0,0 +1,303 @@
+        }
+        if (mWakeLock != null) {
+            mWakeLock.acquire();
+        }
+
+        super.onResume();
+        Log.d(LOGTAG, "onPause");
+    }
+
+    // keep the device's screen turned on and bright.
+    private void keepScreenOn() {
+        if( android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN_MR1 ) {
+            getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+        }
+        else {
+            final PowerManager pm = (PowerManager) getSystemService(Context.POWER_SERVICE);
+            mWakeLock = pm.newWakeLock(PowerManager.SCREEN_DIM_WAKE_LOCK, getClass().getName());
+        }
+    }
+
+    // Dim toolbar and make the view fullscreen
+    private void setFullScreen()
+    {
+        if( android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
+            setFullScreenAndroid44();
+        }
+        else if( android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.ICE_CREAM_SANDWICH ) {
+            setFullScreenAndroid4();
+        }
+    }
+
+    @TargetApi(14)
+    private void setFullScreenAndroid4()
+    {
+        getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_VISIBLE);
+        getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LOW_PROFILE);
+    }
+
+    @TargetApi(19)
+    private void setFullScreenAndroid44()
+    {
+        View decorView = getWindow().getDecorView();
+        decorView.setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+                | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+                | View.SYSTEM_UI_FLAG_FULLSCREEN
+                | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+    }
+
+    // This is here to make the toolbar autohide after 2 seconds of being touched
+    private void addFullScreenListener(final View view)
+    {
+        view.setOnSystemUiVisibilityChangeListener(
+                new View.OnSystemUiVisibilityChangeListener() {
+                    public void onSystemUiVisibilityChange(int visibility) {
+                        if (visibility == 0) {
+                            Runnable rehideRunnable = new Runnable() {
+                                public void run() {
+                                    setFullScreen();
+                                }
+                            };
+                            Handler rehideHandler = new Handler();
+                            rehideHandler.postDelayed(rehideRunnable, 2000);
+                        }
+                    }
+                });
+    }
+
+    private String loadAsset(String file) {
+        InputStream is = null;
+        InputStreamReader isr = null;
+        BufferedReader reader = null;
+        try {
+            is = getAssets().open(file);
+            isr = new InputStreamReader(is);
+            reader = new BufferedReader(isr);
```

Comment:
nit: `reader = new BufferedReadre(new InputStreamReader(is));` so you won't have to deal manually with isr.